crates/sats/src/typespace.rs

```diff
@@ -183,6 +183,55 @@ impl Typespace {
         }
         Ok(())
     }
+
+    /// Validates that the typespace is in "nominal normal form".
+    ///
+    /// Nominal normal form is intended for use when generating code in languages with nominal type systems.
+    ///
+    /// In nominal normal form, all `Sum` and `Product` types in the typespace must be referred to via `Ref`s. That is, all `Sum`, `Product`, `Array`, or `Map` types must contain only `Ref`s and `Builtin`s.
+    /// The exception is empty sums and products (the never and unit types), which are allowed
```

Comment:
I think we currently also allow `Option<T>` i.e. `Sum { some(T), none(Unit) }`, `Identity` i.e. `Product { __identity_bytes(Array { U8 }) }` and `Address` i.e. `Product { __address_bytes(Array { U8 }) }`.

Reply:
I remember `Unit` (empty product) also being treated differently somewhere.